aiofreepybox/api/player.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Player
+    """
+
+    def __init__(self, access, player_api_version=None):
+        self._access = access
+        self._player_api_version = (
+            _DEFAULT_PLAYER_API_VERSION
+            if player_api_version is None
+            else player_api_version
+        )
+
+    media_control_seek_args = {"seek_position": 0, "type": "seek_position"}
+
+    media_control_stream = {"quality": "", "source": ""}
+
+    media_control_stream_args = {"stream": media_control_stream, "type": "stream"}
+
+    media_control_track_args = {"track_id": 0, "type": "track_id"}
+
+    media_control_command = [
+        "play",
+        "pause",
+        "play_pause",
+        "stop",
+        "next",
+        "prev",
+        "seek_forward",
+        "seek_backward",
+        "seek_to",
+        "repeat_all",
+        "repeat_one",
+        "repeat_off",
+        "repeat_toggle",
+        "shuffle_on",
+        "shuffle_off",
+        "shuffle_toggle",
+        "record",
+        "record_stop",
+        "select_audio_track",
+        "select_srt_track",
+        "select_stream",
+    ]
+
+    media_control_data_schema = {
+        "args": media_control_stream_args,
+        "cmd": media_control_command[0],
+    }
+
+    async def get_player_status(self, player_id):
+        """
+        Get player status
+
+        player_id : `int`
+        """
+        return await self._access.get(
+            f"player/{player_id}/api/{self._player_api_version}/status/"
+        )
+
+    async def get_player_volume(self, player_id):
+        """
+        Get player volume
+
+        player_id : `int`
+        """
+        return await self._access.get(
+            f"player/{player_id}/api/{self._player_api_version}/control/volume"
+        )
+
+    async def get_players(self):
+        """
+        Get players
+        """
+        return await self._access.get("player")
+
+    async def open_player_url(self, player_id, player_url_data=None):
+        """
+        Open player url
+
+        player_id : `int`
+        player_url_data : `str` or `dict`
+            , Default to `None`
+        """
+
+        if player_url_data is None:
+            return
+        player_url_data_schema = dict
+        if type(player_url_data) == type({}):
+            player_url_data_schema = player_url_data
+        else:
+            player_url_data_schema["url"] = player_url_data
+
+        await self._access.post(
+            f"player/{player_id}/api/{self._player_api_version}/control/open",
+            player_url_data_schema,
+        )
+
+    async def send_media_control(self, player_id, media_control_data):
+        """
+        Send media control
+
+        player_id : `int`
+        media_control_data : `dict`
+        """
+        await self._access.post(
+            f"player/{player_id}/api/{self._player_api_version}/control/mediactrl",
+            media_control_data,
+        )
+
+    async def set_player_volume(self, player_id, mute=None, volume=None):
```

Comment:
This approach is better IMHO. Here, instead of having a single function accepting either a dict or two booleans, you defined two functions, one that is consistent with the other functions of the library (`update_player_volume`, which should maybe be named `set_player_volume` or `edit_player_volume` as these are the verbs used in other places), and another one that is just a helper to provide a nicer interface. If you could refactor `open_player_url` to follow the same approach, that would be better I think.

Reply:
I agree, i'm still a bit confused on the whole naming schemes. Update, set and edit all pretty much means the same for a freebox, you can argue that set would only imply adding new data or update and edit only means changing existing data, and do the same thing (so which one do i choose ?), but when everything collides more or less depending on the file, it's a mess, i'll adjust on a case by case basis

Reply:
I don't really know what to do with the player_volume names, i'll let you decide which path you want to go (use the force luke ! (sorry :)))

Reply:
Open does not fit into the get, set, del model anyway, so i'll try to apply the following rule from now on:
get,set,del will only take dicts in and out
update will mostly take strs but could take dicts on occasion.
thoughts ?

Reply:
Seems like a sensible convention 👍